lib/generator/js.rb

```diff
@@ -271,6 +273,18 @@ def js_for_data_criteria(force_sources=nil)
         HQMF2JS::Generator.render_template('data_criteria', {'all_criteria' => @doc.specific_occurrence_source_data_criteria(force_sources).concat(@doc.all_data_criteria), 'measure_period' => @doc.measure_period})
       end
       
+      def js_for_variables()
+        variables_js = ""
+        variables_js += "hqmfjs.VARIABLES = function(patient, initialSpecificContext) {\n" 
+        @doc.source_data_criteria.each do |criteria|
+          if criteria.type == :derived
```

Comment:
Is this a correct and sufficient test for whether a criteria is a variable? Specifically, a non-specific-occurrence variable?